Dart/src/com/jetbrains/lang/dart/analyzer/DartAnalysisServerService.java

```diff
@@ -331,6 +345,148 @@ public LibraryDependenciesResult(@Nullable final String[] libraries,
     }
   }
 
+  private void configureImportedLibraries(@NotNull final Collection<String> rootsToAddToLib) {
+    runInEventDispatchThread(new Runnable() {
+      @Override
+      public void run() {
+        ApplicationManager.getApplication().runWriteAction(new Runnable() {
+          public void run() {
+            doConfigureImportedLibraries(rootsToAddToLib);
+          }
+        });
+      }
+    });
+  }
+
+  private static void runInEventDispatchThread(final Runnable runnable) {
+    try {
+      if (SwingUtilities.isEventDispatchThread()) {
+        runnable.run();
+      }
+      else {
+        ApplicationManager.getApplication().invokeAndWait(new Runnable() {
+          public void run() {
+            runnable.run();
+          }
+        }, ModalityState.defaultModalityState());
+      }
+    }
+    catch (Exception e) {
+      LOG.warn(e);
+    }
+  }
+
+  private void doConfigureImportedLibraries(@NotNull final Collection<String> libraries) {
+    for (final Project project : myRootsHandler.getTrackedProjects()) {
```

Comment:
Is there any chance to get information what library file is used in what project?
For example: let us have 2 projects with completely different set of dependencies. If we open any one of them - 'Dart Packages' library for it is configured correctly. If we open both projects simultaneously then we get both projects reconfigured so that Dart Packages library of each of them contains a superset.
**And even worse:** for each individual project `fileIndex.isInContent(vFile)` will return `false` for all files from another project. As a result Dart Packages lib for 1st project will include all content roots of the 2nd project and vice versa. We need to sort it out somehow.